.github/workflows/nightlies.yml

```diff
@@ -58,38 +58,8 @@ jobs:
           mv *.whl dist/
           ls -l dist/
 
-        # N.B. anaconda-client is only maintained on the main channel
-      - name: Install anaconda-client
-        uses: mamba-org/setup-micromamba@v1
-        with:
-          environment-name: nightlies
-          create-args: anaconda-client=1.10.0
-          condarc: |
-            channels:
-              - main
-
       - name: Upload wheels to Anaconda Cloud as nightlies
-        run: |
-          anaconda --token ${{ secrets.ANACONDA_ORG_UPLOAD_TOKEN }} upload \
-            --user scientific-python-nightly-wheels \
-            --skip-existing \
-            dist/matplotlib-*.whl
-
-      - name: Remove old uploads to save space
-        run: |
-          N_LATEST_UPLOADS=5
-
-          # Remove all _but_ the last "${N_LATEST_UPLOADS}" package versions
-          # N.B.: `anaconda show` places the newest packages at the bottom of
-          # the output of the 'Versions' section and package versions are
-          # preceded with a '   + '.
-          anaconda show scientific-python-nightly-wheels/matplotlib &> >(grep '+') | \
-              sed 's/.* + //' | \
-              head --lines "-${N_LATEST_UPLOADS}" > remove-package-versions.txt
-
-          while LANG=C IFS= read -r package_version ; do
-              echo "Removing scientific-python-nightly-wheels/matplotlib/${package_version}"
-              anaconda --token ${{ secrets.ANACONDA_ORG_UPLOAD_TOKEN }} remove \
-                --force \
-                "scientific-python-nightly-wheels/matplotlib/${package_version}"
-          done <remove-package-versions.txt
+        uses: scientific-python/upload-nightly-action@8f0394fd2aa0c85d7364a9958652e8994e06b23c # 0.1.0
```

Comment:
Is this not tagged? Why not use the tag instead of putting it in a comment?

Reply:
@QuLogic It is tagged and that can be done if you'd prefer. c.f. https://github.com/scientific-python/upload-nightly-action/issues/7#issuecomment-1564846744 and @henryiii's comment for the motivation for this as it is now.

Reply:
I think the general principle is that tags can be made to point to a different location, but SHAs cannot (`git checkout` will warn and go to the commit (not the tag)... and fail if the commit does not exist)

So, under the threat model of the tagging authority of upstream being compromised, it is safer to use the SHA.

Reply:
Yes, thanks for the concise summary @ksunden. Dependabot also knows how to handle this and also update the comment, so as `matplotlib` already has

https://github.com/matplotlib/matplotlib/blob/bfaa6eb677b9c56cafb6a99d6897c9d0cd9d4210/.github/dependabot.yml#L1-L6

you can let Dependabot take care of everything associated to updating the commit SHA.

Reply:
To be clear, I don't do this for all GitHub Actions, but when it comes to uploading distributions it seems like this is a good time to be overly cautious. I am willing to admit that I also think it is just good to see these patterns exist in more places.

Reply:
We should perhaps change `cibuildwheel.yml` to do the same then, because those are what we actually publish...

Reply:
Opened PR #26025 for this.

Reply:
Thanks for the fast review and merge on PR #26025 @QuLogic!